integration-tests/profiler/profiler.spec.js

```diff
@@ -407,9 +424,10 @@ describe('profiler', () => {
           }
         }
       }
-      // Need to have a total of 9 different spans, with 3 different root spans
-      // and 3 different endpoints.
+      // Need to have a total of 9 different spans, with 9 different async IDs,
+      // 3 different root spans, and 3 different endpoints.
       assert.equal(spans.size, 9, encoded)
+      assert.equal(asyncIds.size, 9, encoded)
```

Comment:
I guess this could even be zero due to the IDs sometimes missing?

Reply:
We collect multiple samples for each span – we'll have a total of 9 spans, each span runs for ~100ms, and we sample at 99Hz, so statistically each span will have ~9.9 samples (I might increase the span execution time to 102ms which'd bring the expected sample value a tad over 10.) So we'd need to fail to capture the async ID in all 9-10 samples for a span in order for a particular async ID to be completely absent from this set. 

Unfortunately, this is [exactly what we see on Ubuntu profiler tests now](https://github.com/DataDog/dd-trace-js/actions/runs/14227845270/job/39872387872?pr=5524): it observes 2-3 async IDs. Not sure what to make of it, but I'll be inspecting this by running some tests locally on an arm Ubuntu VM to try to reproduce.

Reply:
On a closer look at the [profiling GH workflow](https://github.com/DataDog/dd-trace-js/blob/master/.github/workflows/profiling.yml) though it turns out that we only run profiling tests with `active-lts` version of Node on macOS, but additionally run them with `oldest-maintenance-lts` and `newest-maintenance-lts` on Linux.

Oldest maintenance is 18.20.8 and for it the test indeed fails even on my local macOS. It also fails for 20.19.0, but it passes for 22.14.0. I guess one way to fix the issue is to only enable the feature only on Node 22 and above. It's an interesting question though why does `node::AsyncHooksGetExecutionAsyncId()` work reliably on Node 22 but not on earlier ones. Do you maybe have some information on this?

Reply:
Hm, I bisected on minor node versions and 22.10.0 is the first one where the test reliably passes, and 22.9.0 the last one where it reliably fails. The functions that execute here: `AsyncHooksGetExecutionAsyncId` and its callees  `Environment::GetCurrent` and `Environment::execution_async_id` don't seem like they had a code change in years. Except for https://github.com/nodejs/node/commit/d6bcc448296af0d57e87eff3762a8cea22602cb8 which did indeed go into 22.10.0 but I'd be surprised if that was the one that made the difference.